lib/tasks/demo.rake

```diff
@@ -24,19 +36,163 @@ desc 'Backup the database'
   desc 'Generate N many more monitorees based on existing data'
   task create_bulk_data: :environment do
     raise 'This task is only for use in a development environment' unless Rails.env == 'development' || ENV['DISABLE_DATABASE_ENVIRONMENT_CHECK']
+
     num_patients = (ENV['COUNT'] || 100000).to_i
     num_threads = (ENV['THREADS'] || 8).to_i
 
-    duplicateable = Patient.where('responder_id = id').pluck(:id)
-    threads = []
-    (1..num_threads).each do |x|
-      threads << Thread.new do  (1..(num_patients/num_threads)).each {|x| deep_duplicate_patient(Patient.find(duplicateable.sample)) } end
+    patient_ids = Patient.where('patients.responder_id = patients.id').limit(num_patients).pluck(:id)
+
+    pids = []
+
+    ::ActiveRecord::Base.clear_all_connections!
+    fork_num = 1
+    patient_ids.each_slice(patient_ids.size / num_threads).each do |slice_ids|
+      pids << fork do
+        t1 = Time.now
+        ::ActiveRecord::Base.establish_connection
+
+        num_to_create = num_patients / num_threads
+        num_created = 0
+
+        num_to_create.times do
+          deep_duplicate_patient(Patient.find(slice_ids.sample))
+          num_created += 1
+          print "\r#{(num_created / (Time.now - t1)).truncate(2)} p/s"
+        end
+
+        elapsed = Time.now - t1
+        puts "\n\nFork #{fork_num} has created #{num_created} patients in #{elapsed} seconds. (#{(num_created / elapsed).truncate(2)} patients / sec)"
+      ensure
+        ::ActiveRecord::Base.clear_all_connections!
+        Process.exit! true
+      end
+      fork_num += 1
+    end
+
+    pids.each { |pid| Process.waitpid(pid, 0)  }
+    puts "\nDone!"
+  end
+
+  desc 'Configure the users in the database for performance testing'
+  task setup_performance_test_users: :environment do
```

Comment:
Why are there two instances of this task in this file?

Reply:
i messed up a merge or rebase earlier